sotodlib/core/context.py

```diff
@@ -329,6 +340,97 @@ def get_obs(self,
                         _det_info.move(k, None)
                 meta.det_info.merge(_det_info)
             aman.merge(meta)
+
+        # Deal with special channels, if they exist.
+        # We will merge the dets and tdets axes
+        # And merge the tdets signal into the dets signal, band, and channels.
+        # nans will be inserted into every other dataset assigned to the dets axis.
+        if special_channels and reindex_dets:
+            # Check for special channels (tones).
+            if 'tones' not in aman:
+                logger.info('"tones" not found in aman, no special channels to reindex!')
+
+            # Check there is det_info to reindex.
+            elif 'det_info' not in aman:
+                logger.error('"det_info" not found in aman, no dets to reindex!')
+
+            # Both tones and det_info exist.
+            else:
+                # Grab all band and channel info for dets + tdets
+                det_bands = aman.det_info.smurf.band
+                det_channels = aman.det_info.smurf.channel
+                tdet_bands = aman.tones.band
+                tdet_channels = aman.tones.channel
+    
+                # Create a sorted array of dets + tdets
+                special_band_ch = [(b, c) for b, c in zip(tdet_bands, tdet_channels)]
+                normal_band_ch = [(b, c) for b, c in zip(det_bands, det_channels)]
+                band_ch = np.array(sorted(normal_band_ch + special_band_ch))
+    
+                # Grab the det idxs from the det band + channels
+                det_indexes = np.zeros(len(band_ch)) * np.nan
+                for i, (b, c) in enumerate(band_ch):
+                    w = np.where((det_bands == b) & (det_channels == c))[0]
+                    if len(w) == 0:
+                        continue
+    
+                    det_indexes[i] = w[0]
+    
+                # Grab the tdet idxs from the tdet band + channels
+                tdet_indexes = np.full(np.nan, len(band_ch))
```

Comment:
Sorry I made wrong suggestion. This should be `np.full(len(band_ch), np.nan)`. this line raise error now,
Same for L371.